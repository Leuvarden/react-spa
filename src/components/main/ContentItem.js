import React, { Component } from 'react';
import PropTypes from 'prop-types';

class ContentItem extends Component {

    render () {
        const props = this.props;

        return (
            <figure className="content-item" onClick={() => props.updateFilmPanel(props)}>
                <img src={props.img} className="content-item_img"></img>
                <figcaption className="content-item__description">
                    <span className="content-item_year">
                        {/* todo: move it smwhere else */}
                        {props.date.slice(0, 4)}
                    </span>
                    <span className="content-item_title"><b>{props.title}</b></span>
                    <span className="content-item_genre">
                        {/* todo: move it smwhere else */}
                        {props.genres.length > 1 ? `${props.genres[0]} & ${props.genres[1]}` : props.genres[0]}
                    </span>
                </figcaption>
            </figure>
        );
    }
}

ContentItem.propTypes = {
    genres: PropTypes.array,
    updateFilmPanel: PropTypes.func,
    title: PropTypes.string,
    img: PropTypes.string,
    date: PropTypes.string
};

export default ContentItem;