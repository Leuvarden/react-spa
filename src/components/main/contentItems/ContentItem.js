import React from 'react';
import PropTypes from 'prop-types';
import ContentLink from './../../../containers/ContentLink';
import ErrorBoundary from './../../common/Error';
import ContentErrorItem from './ContentErrorItem';
import ContentItemCaption from './ContentItemCaption';

const ContentItem = ({movie}) => {
    return (
        <ErrorBoundary 
            showOnError={ContentErrorItem()}>    
            <figure 
                className="content-item"
                onClick= {() => { document.body.scrollTop = 0; document.documentElement.scrollTop = 0;} }
            >

                <ContentLink id={movie.id}>
                    <img src={movie.poster_path} className="content-item_img" />
                </ContentLink>

                <ContentItemCaption movie={movie}/>
            </figure>
        </ErrorBoundary>
    );
};

ContentItem.propTypes = {
    movie: PropTypes.object
};

export default ContentItem;