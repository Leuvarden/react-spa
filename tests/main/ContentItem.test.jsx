import React from 'react';
import ContentItem from './../../src/components/main/ContentItem.js';

/* eslint-disable no-undef */
describe ('ContentItem', () => {

    const defaultProps = {
        genres: [ 'Adventure', 'Science'],
        img: 'poster_path',
        date: '2014-05-02',
        title:'Pulp Fiction',
        updateFilmPanel: jest.fn()
    };

    it('renders as expected', () => {
        const item = render(
            <ContentItem {...defaultProps}/>
        );
    
        expect(item).toMatchSnapshot();
    });

    it('receives handler', () => {
        const item = shallow(
            <ContentItem {...defaultProps}/>
        );

        item.find('figure').first().simulate('click');
    
        expect(defaultProps.updateFilmPanel).toHaveBeenCalledTimes(1);
    });

});