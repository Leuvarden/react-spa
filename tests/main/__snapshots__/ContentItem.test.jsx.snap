// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentItem renders as expected 1`] = `
<figure
  className="content-item"
  onClick={[Function]}
>
  <img
    className="content-item_img"
    src="poster_path"
  />
  <figcaption
    className="content-item__description"
  >
    <span
      className="content-item_year"
    >
      2014
    </span>
    <span
      className="content-item_title"
    >
      <b>
        Pulp Fiction
      </b>
    </span>
    <span
      className="content-item_genre"
    >
      Adventure & Science
    </span>
  </figcaption>
</figure>
`;
